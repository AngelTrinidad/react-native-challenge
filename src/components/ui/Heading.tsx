import React from "react";
import { StyleSheet } from "react-native";
import Text, { TextProps } from "./Text";

const Heading: React.FC<TextProps> = ({ children, ...rest }) => {
  return (
    <Text style={styles.heading} {...rest}>
      {children}
    </Text>
  );
};

const styles = StyleSheet.create({
  heading: {
    fontSize: 24,
    fontWeight: "700",
  },
});

export default React.memo(Heading);
