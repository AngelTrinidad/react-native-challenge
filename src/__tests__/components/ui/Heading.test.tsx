import React from "react";
import renderer from "react-test-renderer";
import Heading from "../../../components/ui/Heading";

describe("Test Heading component", () => {
  it("should render Heading", () => {
    const { toJSON } = renderer.create(<Heading>Testing</Heading>);
    expect(toJSON()).toMatchSnapshot();
  });
});
