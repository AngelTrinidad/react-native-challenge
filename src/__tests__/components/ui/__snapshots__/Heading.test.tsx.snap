// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Heading component should render Heading 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#0A132C",
        "fontSize": 14,
        "fontWeight": "400",
      },
      Object {
        "textAlign": undefined,
      },
      Array [
        Object {
          "fontSize": 24,
          "fontWeight": "700",
        },
        undefined,
      ],
    ]
  }
>
  Testing
</Text>
`;
